producer-c/producer-cloudwatch-integ/jobs/runner.groovy

```diff
@@ -0,0 +1,216 @@
+import jenkins.model.*
+import org.jenkinsci.plugins.workflow.steps.FlowInterruptedException
+
+START_TIMESTAMP = new Date().getTime()
+HAS_ERROR = false
+
+RUNNING_NODES=0
+
+CREDENTIALS = [
+    [
+        $class: 'AmazonWebServicesCredentialsBinding', 
+        accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+        credentialsId: 'SDK_CANARY_CREDS',
+        secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+    ]
+]
+
+def buildProducer() {
+  sh  """
+    cd $WORKSPACE/producer-c/producer-cloudwatch-integ && 
+    mkdir -p build
+    cd build && 
+    sudo cmake .. && 
```

Comment:
Why do we need sudo? This scares me because this means that it can mutate the node's system.

Reply:
This is because the AWS CPP SDK related items get installed in root. Maybe I can install them in CMAKE_INSTALL_PREFIX instead and this sudo can go then.